frontend/src/cljs/imas_seamap/map/events.cljs

```diff
@@ -537,6 +537,21 @@
      :put-hash (encode-state db)
      :dispatch [:map/popup-closed]}))
 
+(defn add-layer-below-layer
+  "Instead of adding the layer to the end of the list, it instead adds it just
+   below the target layer. Does nothing if the layer is already added."
+  [{:keys [db]} [_ layer target-layer]]
+  (let [active-layers (get-in db [:map :active-layers])
+        index         (first (index-of #(= % target-layer) active-layers))
+        below         (subvec active-layers 0 index)
+        above         (subvec active-layers index)
+        db            (if-not ((set active-layers) layer)
+                        (assoc-in db [:map :active-layers] (vec (concat below [layer] above)))
+                        db)]
+    {:db       db
+     :put-hash (encode-state db)
+     :dispatch [:map/popup-closed]}))
+
```

Comment:
I suppose the main thing I wonder is: is it simpler/clearer to just make `:add-layer` take an optional arg, for example?  (or something along those lines)

Reply:
Yes, that's a good idea. Definitely worth it to not pollute the ns!

Reply:
Done!